src/lib/components/table/cells/pile-cell.svelte

```diff
@@ -0,0 +1,56 @@
+<script lang="ts" context="module">
+  export interface PileComponentProps {
+    components: PileComponent[];
+    maxItems: number;
+  }
```

Comment:
You can use the `ComponentProps` generic to infer the props of a component, and since PileCell just "forwards" props to the component, it shouldn't re-define the props. Especially because `PileCell` types its props as `any`, if `Pile`'s props ever change, like this the compiler wouldn't be able to catch it. If the parent just types the props it passes to `PileCell` as `ComponentProps` of `Pile`, it'd validate it properly.